### PR TITLE
Add Pilot Logger classes with unit tests

### PR DESCRIPTION
**Pull request by wkrzemien**
Pilot Logging system:
PilotLogger is used in the extended version (ExtendedLogger) of the Logger class, which is used by the pilot commands. The PilotLogger is by default off and the ExtendedLogger acts in exactly the same manner as the original Logger class.
The PilotLogger can send messages about the current status of the installation/configuration etc. to RabbitMQ server from which it will be retrieved by the consumer process and stored into Pilot DB
The messages contain timestamps, status information and the pilotUUID (not pilotID!) by which we can identify even pilots which dont have the pilotID assigned.
In addition, the PilotLogger is storing some of the messages locally just before checking the connection to the RabbitMQ server, if the connection is down, the messages are queued if the connection is back the whole set of messages are sent.
The current version uses STOMP protocol and ssl certificates for authentication

Added getUniqueIdFromOS to set the uuid depending on the OS e.g. CREAM, GRID, VM et

### CODE REVIEW

**Comment by fstagni** [comment]
Alright, so anyway even before this, we need to add the basic pilot files: dirac-pilot.py, pilotTools.py, pilotCommands.py. Can you please open a PR for that?

File: PilotLogger/PilotLogger.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+import stomp
+import sys
+import Queue
+#Since PilotLogger can be used without DIRAC relaying only on PilotLoggerTools, we
+#check two possiblities:
+try:
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import generateDict, encodeMessage
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import generateTimeStamp
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import isMessageFormatCorrect
+  from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import readPilotLoggerConfigFile
```
Comment (fstagni) [inline PilotLogger/PilotLogger.py]: These ones become useless now

Comment (wkrzemien) [inline PilotLogger/PilotLogger.py]: Fixed

**Comment by wkrzemien** [comment]
Which version of dirac-pilot.py, pilotTools.py, pilotCommands.py should I use?

File: PilotLogger/test/PilotLoggerToolsTests.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import unittest
+import mock
+import json
+import os
+from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import generateDict, encodeMessage
+from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import decodeMessage, isMessageFormatCorrect
+from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import getUniqueIDAndSaveToFile
+from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import createPilotLoggerConfigFile
+from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import readPilotLoggerConfigFile
+from DIRAC.WorkloadManagementSystem.PilotAgent.PilotLogger.PilotLoggerTools import getUniqueIDFromOS
```
Comment (fstagni) [inline PilotLogger/test/PilotLoggerToolsTests.py]: Not importing DIRAC

**Comment by fstagni** [comment]
import from branch rel-v6r15.

**Comment by fstagni** [comment]
Also, please, import (or update) the dotfiles: gitignore, pylintrc, pytest.ini.

requirements.txt should also be created, but with strictly only the python packages needed by the pilot here.

**Comment by fstagni** [comment]
Please, also:
- make the README an rst file, write something meaningful in it.
- and the docs and tests directories (empty for now)

**Comment by wkrzemien** [comment]
Ok. But do you want  all tests in one separate directory or tests per module like in DIRAC e.g. separate tests dir for PilotLogger classes, separate for PilotTools etc.

**Comment by wkrzemien** [comment]
Ok, nevermind I checked how it is done in DIRAC

**Comment by fstagni** [comment]
Since we'll most probably won't have many packages, it's probably OK to 
put every test inside the tests directory

File: PilotLogger/test/PilotLoggerToolsTests.py (hunk 2 above)
Comment (wkrzemien) [inline PilotLogger/test/PilotLoggerToolsTests.py]: Fixed

**Pull request closed without merging**